pkg/release/release.go

```diff
@@ -333,3 +373,48 @@ func (r *Release) podName(namespace, labelSelector string) (string, error) {
 	pod := pods.Items[0]
 	return pod.Name, nil
 }
+
+func (r *Release) pullTarball(chart string, version Version) (string, error) {
+	if version.isChannel {
+		tarball, err := r.apprClient.PullChartTarball(chart, version.String())
+		if err != nil {
+			return "", microerror.Mask(err)
+		}
+
+		return tarball, nil
+	}
+
+	tarball, err := r.apprClient.PullChartTarballFromRelease(chart, version.String())
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+
+	return tarball, nil
+}
+
+func (r *Release) waitForConditions(ctx context.Context, conditions ...func() error) error {
```

Comment:
I think about exposing it and removing conditions from `Install` & `Update`. BTW `conditions` in `Update` aren't checked at all so maybe we don't even have to expose that ATM. Let's see.